Data/Yaml/TH.hs

```diff
@@ -0,0 +1,39 @@
+{-# LANGUAGE LambdaCase          #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TemplateHaskell     #-}
+
+module Data.Yaml.TH
+  ( -- * Decoding
+    decodeFile
+    -- * Re-exports from "Data.Yaml"
+  , Value (..)
+  , Parser
+  , Object
+  , Array
+  , object
+  , array
+  , (.=)
+  , (.:)
+  , (.:?)
+  , (.!=)
+  , FromJSON (..)
+  ) where
+
+import Data.Yaml hiding (decodeFile)
+import Language.Haskell.TH
+import Language.Haskell.TH.Syntax
+
+-- | Decode a @yaml@ file at compile time.
+--
+-- @
+-- {-# LANGUAGE TemplateHaskell #-}
+--
+-- config :: Config
+-- config = $$('decodeFile' "config.yaml")
+-- @
```

Comment:
Can you add a `@since` comment, bump the minor version, and add a note to the `ChangeLog.md`?